### DISCUSSION TITLE
Metadata is missing in spec when using `Depends()` to get query params

### DISCUSSION THREAD

**Discussion by chbndrhnns** [fastapi/fastapi#9617] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi as f
import pydantic as p
import pytest

app = f.FastAPI()


class Fields:
    param_qry = f.Query(None, description="abc")


class Qry(p.BaseModel):
    param: str = p.Field(description=Fields.param_qry.description)


@app.get("/direct")
async def direct(param: str = Fields.param_qry):
    return {"param": param}


@app.get("/depends")
async def depends(params: Qry = f.Depends()):
    return {"param": params.param}


@pytest.mark.parametrize(
    "endpoint",
    [
        "/direct",
        "/depends",
    ],
)
async def test(endpoint):
    spec = app.openapi()
    assert spec["paths"][endpoint]["get"]["parameters"][0]["description"] == "abc"
```

### Description

I am using the concept from [advanced dependencies](https://fastapi.tiangolo.com/advanced/advanced-dependencies/) to get query parameters.

So I expect both tests to pass. However, the test for `/depends` fails as the description is missing from the schema.

Is there a problem in my thinking or a problem in FastAPI?

### Operating System

macOS

### FastAPI Version

0.96.0

### Python Version

3.9.6

**Comment by kevinrosa** [comment]
Any updates on this? I'm experiencing the same issue and have not found a workaround to use classes as dependencies while preserving metadata for docs.  I'll paste my minimal working example which is effectively the same but written a bit different so may catch someone's attention:

```python
from pydantic import BaseModel
from fastapi import FastAPI, Query, Depends

app = FastAPI(
    title="Metadata test",
)

# This method shows the metadata in the docs...
@app.get("/parameters_direct")
def parameters_direct(
    region: str = Query(example="boston", description="Name of requested model region."),
    api_key: str = Query(description="Your personal API key for authentication."),
):
    pass

# This method does not show the metadata in the docs...
class RequestModel(BaseModel):
    region: str = Query(example="boston", description="Name of requested model region.")
    api_key: str = Query(description="Your personal API key for authentication.")

@app.get("/parameters_dependency")
def parameters_dependency(request: RequestModel = Depends()):  # Use Depends to treat it as query parameter dependencies (instead of expecting a JSON body)
    pass
```
<img width="1193" alt="Screen Shot 2023-10-26 at 4 52 39 PM" src="https://github.com/tiangolo/fastapi/assets/13137098/08c120d8-ad48-4a60-bc9d-a0da5271016e">
<img width="1209" alt="Screen Shot 2023-10-26 at 4 52 57 PM" src="https://github.com/tiangolo/fastapi/assets/13137098/9a9fb390-228d-4ef4-81fc-d7a45982ade0">

**Comment by namratasharmaprecisely** [comment]
Was anyone able to figure this out? :') 
Was able to get my parameter model metadata to show in the docs when I do this:
```
@router.post("/endpoint")
    async def prep_endpoint(param: ParamModel, file: UploadFile = File(..., description="Upload file"))
```
Where ParamModel is defined as:
```
from pydantic import BaseModel, Field

class ParamModel(BaseModel):
    """Model representing param metadata."""
    id: str = Field(..., example="some_val", description="id val")
    ...
```


BUT I want to use Depends! Like this:
```
@router.post("/endpoint")
    async def prep_endpoint(param: ParamModel = Depends(), file: UploadFile = File(..., description="Upload file"))
```
OR BETTER yet, like this:
```
@router.post("/endpoint")
    async def prep_endpoint(param: ParamModel = Depends(param_parser), file: UploadFile = File(..., description="Upload file"))
```
Where the param_parser function is:
```
import json
from model import ParamModel

def param_parser(param: str = Form(...)):
      param_dict = json.loads(param)
      return ParamModel(**param_dict)
```
This is especially important for me bc I want my client app to send my param to this endpoint as a json, which means I need to parse it before setting it as my desired ParamModel type.
(This is bc I want to send an UploadFile to this endpoint too, which means my param _has_ to be sent as a json.) 
I'm sure there are workarounds w how I send my param and UploadFile to the endpoint from the client, but the point is, Depends really simplifies my client-side call! 
But using Depends, whether by itself or w a function passed in, means my param model metadata (ex. description, example) doesn't display. In /docs or in /redocs. :') Why oh why!

**Reply by namratasharmaprecisely** [reply]
On the off chance someone's having the same problem...I fixed it by modifying my ParamModel class. 
```
from pydantic import BaseModel, Field

class ParamModel(BaseModel):
    """Model representing param metadata."""
    id: str = Field(..., example="some_val", description="id val")
    
    # This pre-validator handles string conversion of raw data input to dict if needed
    @root_validator(pre=True)
    def parse_string_input(cls, values):
        if isinstance(values, str):
            try:
                return json.loads(values) # Try to parse the string as JSON
            except json.JSONDecodeError:
                raise ValueError("Invalid JSON string provided")
        return values
```
So now I can use ParamModel, run a function on the code automatically (w/o Depends), and the docs still work :D 
```
@router.post("/endpoint")
    async def prep_endpoint(param: ParamModel, file: UploadFile = File(..., description="Upload file"))
```

**Comment by YuriiMotov** [comment, marked as answer]
Since FastAPI 0.115.0 Query Parameter Models are officially supported in FastAPI: https://fastapi.tiangolo.com/tutorial/query-param-models/
